credentials/credentials.go

```diff
@@ -18,24 +18,23 @@ var (
 )
 
 // Setup ensures authentication is set up
-func Setup(endpoint string) (err error) {
+func Setup(endpoint string) (token string, err error) {
```

Comment:
why do you use named return values? No need for that.

Reply:
Unless you set the value in a deferred statement after the `return`

Reply:
It's a stylistic choice. 

I find it makes it clearer what variables are going to be returned.

Reply:
I gotta subscribe to the Dave Cheney ["Avoid named return values" rule](https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values) for practical Go.

Also [A Tour of Go](https://tour.golang.org/basics/7):
>Naked return statements should be used only in short functions, as with the example shown here. They can harm readability in longer functions.

This probably meets the "short function" criteria to be acceptable but care should be taken... or make or your functions much shorter ;)